ph5/utilities/set_deploy_pickup_times.py

```diff
@@ -85,91 +83,199 @@ def get_args():
                         help="The Array_t_xxx.kef file to modify.",
                         metavar="array_kef", required=True)
 
-    parser.add_argument("-d", "--deploy-time", dest="deploy_time",
+    parser.add_argument("-n", "--nickname", default="master.ph5",
+                        type=str, metavar="nickname", required=False,
+                        help="Name of the PH5 file.", dest="nickname")
+
+    parser.add_argument("-p", "--ph5path", action="store", default=".",
+                        help=("Comma separated list of paths to ph5 "
+                              "experiments."),
+                        type=str, metavar="ph5path", required=False,
+                        dest="path")
+
+    parser.add_argument("-dt", "--deploy-time", dest="deploy_time",
                         help="Array deployment time: YYYY:JJJ:HH:MM:SS",
-                        metavar="deploy_time", required=True)
+                        metavar="deploy_time", required=False)
 
-    parser.add_argument("-p", "--pickup-time", dest="pickup_time",
+    parser.add_argument("-pt", "--pickup-time", dest="pickup_time",
                         help="Array pickup time: YYYY:JJJ:HH:MM:SS",
-                        metavar="pickup_time", required=True)
+                        metavar="pickup_time", required=False)
 
-    args = parser.parse_args()
+    parser.add_argument("-ac", "--auto-correct", dest="auto_correct",
+                        help="Assigns the metadata deploy and pickup times"
+                             "to time stamps calculated by the PH5 traces.",
+                        metavar="auto_correct", required=False, default=False)
 
+    args = parser.parse_args()
+    NICKNAME = args.nickname
+    PATH = args.path
     ARRAY_FILE = args.array_kef
     DEPLOY = args.deploy_time
     PICKUP = args.pickup_time
+    AUTO_CORRECT = args.auto_correct
+
 
+def barf(fh, of, dep_time, pu_time, auto):
```

Comment:
Can we change the name of this method to something that is more obvious?

Reply:
Commit 896cee4